fastlane/lib/fastlane/runner.rb

```diff
@@ -197,7 +197,15 @@ def try_switch_to_lane(new_lane, parameters)
 
         # Actually switch lane now
         self.current_lane = new_lane
-        collector.did_launch_action(:lane_switch)
+
+        app_id_guesser = FastlaneCore::AppIdentifierGuesser.new(args: ARGV)
```

Comment:
this could be combined with the same block of code found at line 232 similar to how you did it in `fast_file.rb`
You could also write a small function that could be shared by all :)